core/arch/aarch64/instr_create.h

```diff
@@ -369,8 +369,11 @@
  * they just need to know whether they need to preserve the app's flags, so maybe
  * we can just document that this may not write them.
  */
-#define XINST_CREATE_slr_s(dc, d, s) \
-  instr_create_1dst_2src((dc), OP_lsrv, (d), (d), (s))
+#define XINST_CREATE_slr_s(dc, d, rm_or_imm) \
+  opnd_is_reg(rm_or_imm) ? \
+    instr_create_1dst_2src((dc), OP_lsrv, (d), (d), (rm_or_imm)) : \
+    instr_create_1dst_3src((dc), OP_ubfm, (d), (d), (rm_or_imm), \
```

Comment:
1. Probably, thanks for spotting that!

2. Got rid of them. It seems like aarch64/instr_create.h could do with some cleanup, there seem to be a lot of unnecessary parentheses.

3. Fixed as well, instr_create.h is excluded from the line length style check (https://github.com/DynamoRIO/dynamorio/blob/master/make/style_checks/scripts/rules/line_length_content.tcl#L48), probably because there are plenty of violations on other architectures?

4. We should definitely discuss how to deal with aliases, maybe the mailing list would be a good place? I think it makes live easier for users if we provide macros for aliases that can deal with different operand types properly. Although as you pointed out, `INSTR_CREATE_mov` would be massive.